spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/micrometer/RabbitMessageSenderContext.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.support.micrometer;
+
+import org.springframework.amqp.core.Message;
+
+import io.micrometer.observation.transport.SenderContext;
+
+/**
+ * {@link SenderContext} for {@link Message}s.
+ *
+ * @author Gary Russell
+ * @since 3.0
+ *
+ */
+public class RabbitMessageSenderContext extends SenderContext<Message> {
+
+	private final String beanName;
+
+	private final String destination;
+
+	public RabbitMessageSenderContext(Message message, String beanName, String destination) {
+		super((carrier, key, value) -> message.getMessageProperties().setHeader(key, value));
+		setCarrier(message);
+		this.beanName = beanName;
+		this.destination = destination;
+	}
+
+	public String getBeanName() {
+		return this.beanName;
+	}
+
+	@Override
+	public String getName() {
+		return "spring.rabbit.template";
```

Comment:
I used the name directly on the `DocumentedObservation` impl instead of context.
Not sure what is the proper way to go though...
@marcingrzejszczak ?

Thanks

Reply:
The rule of thumb should be

- always try to use the names from a convention
- if for some reason it doesn't make sense (you would have a lot of conventions cause you have a lot of contexts that don't differ too much) you can set one on a DocumentedObservation